### ISSUE TITLE
theloop.2.rom has been renamed, lets discuss

### ISSUE THREAD

**Issue by kervinck** [PhilThomas/gigatron#1]
In preparation of new ROM versions, we have renamed "theloop.2.rom" to "ROMv1.rom" in the Gigatron  repository. The URLs with this name it it don't work anymore :-(

We intend to keep multiple ROM versions visible in the gigatron-rom repo.
Perhaps (in the README.md) link to the repo itself, so the JS emulator user can pick the preferred ROM?

What we can also propose is that we keep the "latest" ROM version in the Gigatron-rom's under a stable name (but not "theloop.2.rom" :-)...)

Or just follow it here, and update the link once in a while. It is not as if we will have dozens of ROM versions out there..

Lets discuss what will work for you.

**Comment by PhilThomas** [comment]
It is difficult for the js app to load a ROM from github because it is cross origin.  I did figure out how to do it with the github REST API, JSONP, and converting base64 to binary, but it is complicated and inefficient.  It is better to host the ROM images on the gigatron.io server as we did with `theloop.2.rom`.  We would then create a js file that defines a const containing an array of ROMs with names, versions, and URLs that would be referenced from `index.html`.  We could also designate one image as the default in that file.

Will new ROMs be backwards compatible?  i.e. will a loadable image that runs on one ROM version also run on a new version?  Also, will newer ROMs provide additional system calls to loadable images?  If so, then it would be a good idea to be able to tag images with the minimum ROM version required to run them.  That way, the emulator can make sure that a compatible ROM version is loaded when attempting to load an image.

**Comment by kervinck** [comment]
Yes, we will do everything to maintain compatibility with existing GT1 files. There will be new features, but GT1 files that use them are supposed to check for them first: https://forum.gigatron.io/viewtopic.php?f=4&t=44

The remark above is more about the instructions here in the README.md and main.js files: the name "theloop.2.rom" doesn't exist anymore, it has been renamed to "ROMv1.rom"... So that might be confusing to new people.

Soon we will add a minor ROM update, and publish "ROMv2.rom" alongside it. They are compatible, and we're planning to keep both ROM files visible in our repo at all times. It is perfectly fine that the files must be installed as a separate step. I've been thinking, perhaps the simplest is just to rename 'theloop.2.rom' here to "ROMv1.rom", and leave it at that. I have no plans to rename it again :-)

[ BTW, ROM v3 will probably get support for ASCII and matrix keyboards, as we want support for embedded BASIC. ]

**Comment by kervinck** [comment]
BTW somewhat off-topic: have you considered integrating the JS emulator into the main repository (under Contrib/PhilThomas/?). See also https://forum.gigatron.io/viewtopic.php?f=4&t=19
